---
title: "metag-metat"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      message = FALSE,
                      warning = FALSE, 
                      fig.path = "images/markdown-metagt/"
)
```


```{r}
library("funrar")
library("vegan")
library("ape")
library("reshape2")
library("DESeq2")
library("preprocessCore")
library("ggplot2")
library("dplyr")
library(pheatmap)
library(tidyverse)
library(grid)
library(viridis)

# set ggplot theme 
theme_kp <- function() {  # this for all the elements common across plots
  theme_bw() %+replace%
    theme(
      legend.key=element_blank(),
      legend.title = element_blank(),
      legend.text = element_text(size = 12),
      legend.key.size = unit(1.5, 'lines'),
      panel.border = element_rect(color="black",size=1.5, fill = NA),
      
      plot.title = element_text(hjust = 0.05, size = 14),
      axis.text = element_text(size = 14,  color = "black"),
      axis.title = element_text(size = 14, face = "bold", color = "black"),
      
      # formatting for facets
      panel.background = element_blank(),
      strip.background = element_rect(colour="white", fill="white"), #facet formatting
      panel.spacing.x = unit(1.5, "lines"), #facet spacing for x axis
      panel.spacing.y = unit(1.5, "lines"), #facet spacing for x axis
      strip.text.x = element_text(size=12, face="bold"), #facet labels
      strip.text.y = element_text(size=12, face="bold", angle = 270) #facet labels
    )
}
```

```{r metat_pca, include=FALSE}
# PCA ---------------------------------------------------------------------
g_tab = read.table("data/metagt/metaT_combined_0.001perc_removed.txt", sep="\t", header=TRUE, row.names=1)

# Convert all the na's to zeros
g_tab[is.na(g_tab)] = 0

# Remove base_line samples
g_tab = subset(g_tab, Treatment!="Base_Line")

# Remove ribosomal genes
g_tab = subset(g_tab, select = -c(TIGR00001,	TIGR00002,	TIGR00009,	TIGR00012,	TIGR00029,
                                  TIGR00030,	TIGR00059,	TIGR00060,	TIGR00061,	TIGR00062,	TIGR00105,	TIGR00158,	
                                  TIGR00165,	TIGR00166,	TIGR00279,	TIGR00717,	
                                  TIGR00731,	TIGR00855,	TIGR00952,	TIGR00981,	TIGR00982,	TIGR01008,	TIGR01009,	
                                  TIGR01011,	TIGR01012,	TIGR01017,	TIGR01018,	TIGR01020,	TIGR01021,	TIGR01022,	
                                  TIGR01023,	TIGR01024,	TIGR01025,	TIGR01028,	TIGR01029,	TIGR01030,	TIGR01031,	
                                  TIGR01032,	TIGR01038,	TIGR01044,	TIGR01046,	TIGR01049,	TIGR01050,	TIGR01066,	
                                  TIGR01067,	TIGR01071,TIGR01079,	TIGR01080,	TIGR01125,	TIGR01164,	
                                  TIGR01169,	TIGR01170,	TIGR01171,	TIGR01308,TIGR01632,	
                                  TIGR03626,	TIGR03627,	TIGR03628,	TIGR03629,TIGR03631,	TIGR03632,	
                                  TIGR03635,	TIGR03654,	TIGR03953,  TIGR03673))

# Make a matrix that excludes all the metadata and is just numbers with the sample numbers as the row names
NAMES = rownames(g_tab)
g_matrix = g_tab[,7:1145]
rownames(g_matrix) = NAMES
g_matrix = as.matrix(g_matrix)
quantile(g_matrix)

NAMES = rownames(g_tab)
g_sample = g_tab[,1:6]
rownames(g_sample) = NAMES



#
## RELATIVE ABUNDANCE normalization ----------------------------------------

g_rel = make_relative(g_matrix)
merged = merge(g_sample, g_rel, by="row.names")

#

e_distance = vegdist(g_rel, method="euclidean")
principal_coordinates = pcoa(e_distance)

pcoa_plot = data.frame(principal_coordinates$vectors[,])
pcoa_plot_merged = merge(pcoa_plot, g_sample, by="row.names")

# Calculate percent variation explained by PC1, PC2

PC1 <- 100*(principal_coordinates$values$Eigenvalues[1]/sum(principal_coordinates$values$Eigenvalues))
PC2 <- 100*(principal_coordinates$values$Eigenvalues[2]/sum(principal_coordinates$values$Eigenvalues))
PC3 <- 100*(principal_coordinates$values$Eigenvalues[3]/sum(principal_coordinates$values$Eigenvalues))

# Plot PCoA ----
pcoa_plot_merged$Treatment = factor(pcoa_plot_merged$Treatment, levels = c("Drought","Field_Moist","Sat_II","Sat_I"))

gg_metat_pca = 
  ggplot(data=pcoa_plot_merged,aes(x=Axis.1,y=Axis.2))+
  geom_point(aes(fill=factor(Treatment),shape=factor(Site)),size=6,alpha=0.95) + theme_bw()  +
  theme_kp() + 
  stat_ellipse(aes(color=Treatment),size=1.5)+
  labs(x = paste("PC1 - Variation Explained", round(PC1,2),"%"), 
       y = paste("PC2 - Variation Explained", round(PC2,2),"%"),
       title = "metatranscriptomes")+
  scale_fill_manual(values=c("Sat_II"= "#443a83ff","Field_Moist"="#35b779ff",
                             "Drought"="#fde725ff","Sat_I"="grey70"),
                    labels=c("Drought"="drought","Field_Moist"="field moist",
                             "Sat_II"="flood","Sat_I"="time zero saturation"))+
  scale_color_manual(values=c("Sat_II"="#443a83ff","Field_Moist"="#35b779ff",
                              "Drought"="#fde725ff","Sat_I"="grey70"),
                     labels=c("Drought"="drought","Field_Moist"="field moist",
                              "Sat_II"="flood","Sat_I"="time zero saturation"))+
  scale_shape_manual(values=c(21,22,24),
                     labels=c("CPCRW"="Alaska","DWP"="Florida","SR"="Washington"))+
  guides(fill=guide_legend(override.aes=list(shape=21)),color=FALSE)+
  theme(plot.title = element_text(hjust = 0.5, face="bold"))+
  NULL


```

```{r metag_pca, include=FALSE}
# 1. PCA ANALYSIS ------------------------------------------------------------

g_tab = read.table("data/metagt/metaG_combined_0.001perc.txt", sep="\t", header=TRUE, row.names=1)

# Convert all the na's to zeros
g_tab[is.na(g_tab)] = 0

# Remove all ribosomal genes
g_tab = subset(g_tab, select = -c(TIGR00001,	TIGR00002,	TIGR00009,		
                                  TIGR00030,	TIGR00059,	TIGR00060,	TIGR00061,	TIGR00062,	TIGR00105,	TIGR00158,	
                                  TIGR00165,	TIGR00717,	
                                  TIGR00731,	TIGR00855,	TIGR00952,	TIGR00981,	TIGR01009,	
                                  TIGR01011,	TIGR01017,	TIGR01021,	TIGR01022,	
                                  TIGR01023,	TIGR01024,	TIGR01029,	TIGR01030,	TIGR01031,	
                                  TIGR01032,	TIGR01044,	TIGR01049,	TIGR01050,	TIGR01066,	
                                  TIGR01067,	TIGR01071,	TIGR01079,	TIGR01125,	TIGR01164,	
                                  TIGR01169,	TIGR01171,	TIGR01632,	
                                  TIGR03631,	TIGR03632,	
                                  TIGR03635,	TIGR03654,	TIGR03953))

# Make a matrix that excludes all the metadata and is just numbers with the sample numbers as the row names
NAMES = rownames(g_tab)
g_matrix = g_tab[,6:1537]
rownames(g_matrix) = NAMES
g_matrix = as.matrix(g_matrix)
quantile(g_matrix)

NAMES = rownames(g_tab)
g_sample = g_tab[,1:2]
rownames(g_sample) = NAMES

# relative abundance normalization

g_rel = make_relative(g_matrix)
merged = merge(g_sample, g_rel, by="row.names")


# calculate edistances

e_distance = vegdist(g_rel, method="euclidean")
principal_coordinates = pcoa(e_distance)

pcoa_plot = data.frame(principal_coordinates$vectors[,])
pcoa_plot_merged = merge(pcoa_plot, g_sample, by="row.names")

# Calculate percent variation explained by PC1, PC2

PC1 <- 100*(principal_coordinates$values$Eigenvalues[1]/sum(principal_coordinates$values$Eigenvalues))
PC2 <- 100*(principal_coordinates$values$Eigenvalues[2]/sum(principal_coordinates$values$Eigenvalues))
PC3 <- 100*(principal_coordinates$values$Eigenvalues[3]/sum(principal_coordinates$values$Eigenvalues))

# Plot PCoA ----

pcoa_plot_merged$Treatment = factor(pcoa_plot_merged$Treatment, levels = c("Drought","Field_Moist","Sat_II","Sat_I"))

gg_metag_pca = 
  ggplot(data=pcoa_plot_merged,aes(x=Axis.1,y=Axis.2)) + 
  geom_point(aes(fill=factor(Treatment),shape=factor(Site)),  size=6,alpha=0.95) + theme_bw()  +
  theme_kp() + 
  labs(x = paste("PC1 - Variation Explained", round(PC1,2),"%"), 
       y = paste("PC2 - Variation Explained", round(PC2,2),"%"),
       title = "metagenomes") +
  scale_fill_manual(values=c("Sat_II"= "#443a83ff","Field_Moist"="#35b779ff",
                             "Drought"="#fde725ff","Sat_I"="grey70"),
                    labels=c("Drought"="drought","Field_Moist"="field moist",
                             "Sat_II"="flood","Sat_I"="time zero saturation"))+
  scale_color_manual(values=c("Sat_II"="#35b779ff","Field_Moist"="#fde725ff",
                              "Drought"="#443a83ff","Sat_I"="grey70"))+
  scale_shape_manual(values=c(21,22,24),labels=c("CPCRW"="Alaska","DWP"="Florida","SR"="Washington"))+
  guides(fill=guide_legend(override.aes=list(shape=21)))+
  theme(plot.title = element_text(hjust = 0.5, face = "bold"))+
  NULL


```


```{r metagt_pca, fig.height=6, fig.width=12}
ggpubr::ggarrange(gg_metag_pca, gg_metat_pca, ncol=2, nrow=1, common.legend = TRUE, legend="top")

```


----
#### Session Info

date run: `r Sys.Date()`

```{r}
sessionInfo()
```
